prow/github/client.go

```diff
@@ -973,15 +973,55 @@ func (c *Client) AddLabel(org, repo string, number int, label string) error {
 	return err
 }
 
+// LabelNotFound indicates that a label is not attached to an issue. For example, removing a
+// label from an issue, when the issue does not have that label.
+type LabelNotFound struct {
+	Owner, Repo string
+	Number      int
+	Label       string
+}
+
+func (e *LabelNotFound) Error() string {
+	return fmt.Sprintf("label %q does not exist on %s/%s/%d", e.Label, e.Owner, e.Repo, e.Number)
+}
+
+type githubError struct {
+	Message string `json:"message,omitempty"`
+}
+
 func (c *Client) RemoveLabel(org, repo string, number int, label string) error {
 	c.log("RemoveLabel", org, repo, number, label)
-	_, err := c.request(&request{
+	ge := githubError{}
+	code, err := c.request(&request{
 		method: http.MethodDelete,
 		path:   fmt.Sprintf("%s/repos/%s/%s/issues/%d/labels/%s", c.base, org, repo, number, label),
 		// GitHub sometimes returns 200 for this call, which is a bug on their end.
-		exitCodes: []int{200, 204},
-	}, nil)
-	return err
+		// On 404, do not retry. We will inspect and handle this case appropriately.
+		exitCodes: []int{200, 204, 404},
+	}, &ge)
```

Comment:
This means `request` will try to unmarshal any kind of request body into `githubError` (which may yield an Unmarshal error; the test failure I am seeing in https://github.com/kubernetes/test-infra/pull/7258)

Reply:
Nice catch! Thanks for taking care of this in your PR!